azurerm/resource_arm_dev_test_lab_schedule.go

```diff
@@ -0,0 +1,489 @@
+			if err := d.Set("hourly_recurrence", flattenAzureRmDevTestLabScheduleRecurrenceHourly(props.HourlyRecurrence)); err != nil {
+				return fmt.Errorf("Error setting `dailyRecurrence`: %#v", err)
+			}
+		}
+
+		if err := d.Set("notification_settings", flattenAzureRmDevTestLabScheduleNotificationSettings(props.NotificationSettings)); err != nil {
+			return fmt.Errorf("Error setting `notificationSettings`: %#v", err)
+		}
+	}
+
+	flattenAndSetTags(d, resp.Tags)
+
+	return nil
+}
+
+func resourceArmDevTestLabSchedulesDelete(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*ArmClient).vmExtensionClient
+	ctx := meta.(*ArmClient).StopContext
+
+	id, err := parseAzureResourceID(d.Id())
+	if err != nil {
+		return err
+	}
+	resGroup := id.ResourceGroup
+	name := id.Path["schedules"]
+	devTestLabName := id.Path["labs"]
+
+	future, err := client.Delete(ctx, resGroup, devTestLabName, name)
+	if err != nil {
+		return err
+	}
+
+	return future.WaitForCompletionRef(ctx, client.Client)
+}
+
+func expandArmDevTestLabScheduleRecurrenceDaily(d *schema.ResourceData) (*dtl.DayDetails, error) {
+	dailyRecurrenceConfigs := d.Get("daily_recurrence").(*schema.Set).List()
+	dailyRecurrenceConfig := dailyRecurrenceConfigs[0].(map[string]interface{})
+	dailyTime := dailyRecurrenceConfig["time"].(string)
+
+	return &dtl.DayDetails{
+		Time: &dailyTime,
+	}, nil
+
+}
+
+func flattenAzureRmDevTestLabScheduleRecurrenceDaily(dailyRecurrence *dtl.DayDetails) []interface{} {
+	if dailyRecurrence == nil {
+		return []interface{}{}
+	}
+
+	result := make(map[string]interface{})
+
+	if dailyRecurrence.Time != nil {
+		result["time"] = *dailyRecurrence.Time
+	}
+
+	return []interface{}{result}
+}
+
+func expandArmDevTestLabScheduleRecurrenceWeekly(d *schema.ResourceData) (*dtl.WeekDetails, error) {
+	weeklyRecurrenceConfigs := d.Get("weekly_recurrence").(*schema.Set).List()
+	weeklyRecurrenceConfig := weeklyRecurrenceConfigs[0].(map[string]interface{})
+
+	weeklyTime := weeklyRecurrenceConfig["time"].(string)
+
+	weekDays := make([]string, 0)
+	for _, dayItem := range weeklyRecurrenceConfig["week_days"].([]interface{}) {
+		weekDays = append(weekDays, dayItem.(string))
+	}
+
+	return &dtl.WeekDetails{
+		Time:     &weeklyTime,
+		Weekdays: &weekDays,
+	}, nil
+
+}
+
+func flattenAzureRmDevTestLabScheduleRecurrenceWeekly(weeklyRecurrence *dtl.WeekDetails) []interface{} {
+	if weeklyRecurrence == nil {
+		return []interface{}{}
+	}
+
+	result := make(map[string]interface{})
+
+	if weeklyRecurrence.Time != nil {
+		result["time"] = *weeklyRecurrence.Time
+	}
+
+	weekDays := make([]string, 0)
+	if w := weeklyRecurrence.Weekdays; w != nil {
+		weekDays = *w
+	}
+	result["week_days"] = weekDays
+
+	return []interface{}{result}
+}
+
+func expandArmDevTestLabScheduleRecurrenceHourly(d *schema.ResourceData) (*dtl.HourDetails, error) {
+	hourlyRecurrenceConfigs := d.Get("hourly_recurrence").(*schema.Set).List()
```

Comment:
(same here) since this is now a List we can make this:

```suggestion
	hourlyRecurrenceConfigs := d.Get("hourly_recurrence").([]interface{})
```

Reply:
Done.